apm-collector/apm-collector-worker/src/main/java/org/skywalking/apm/collector/worker/httpserver/AbstractGet.java

```diff
@@ -1,56 +1,73 @@
 package org.skywalking.apm.collector.worker.httpserver;
 
 import com.google.gson.JsonObject;
-import org.skywalking.apm.collector.actor.*;
-
+import java.io.IOException;
+import java.util.Map;
 import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.io.IOException;
-import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.skywalking.apm.collector.actor.ClusterWorkerContext;
+import org.skywalking.apm.collector.actor.LocalSyncWorkerRef;
+import org.skywalking.apm.collector.actor.LocalWorkerContext;
+import org.skywalking.apm.collector.actor.Role;
 
 /**
+ * The <code>AbstractGet</code> implementations represent workers, which called by the server to allow a servlet to
+ * handle a GET request.
+ *
+ * <p>verride the {@link #onReceive(Map, JsonObject)} method to support a search service.
+ *
  * @author pengys5
+ * @since v3.0-2017
  */
-public abstract class AbstractGet extends AbstractLocalSyncWorker {
+public abstract class AbstractGet extends AbstractServlet {
 
     protected AbstractGet(Role role, ClusterWorkerContext clusterContext, LocalWorkerContext selfContext) {
         super(role, clusterContext, selfContext);
     }
 
-    @Override
-    final public void onWork(Object request, Object response) throws Exception {
-        Map<String, String[]> parameterMap = (Map<String, String[]>) request;
-        try {
-            onSearch(parameterMap, (JsonObject) response);
-        } catch (Exception e) {
-            ((JsonObject) response).addProperty("isSuccess", false);
-            ((JsonObject) response).addProperty("reason", e.getMessage());
-        }
+    /**
+     * Add final modifier to avoid the subclass override this method.
```

Comment:
`final` signature already meands can't override. Shouldn't add comments about it.